Clear out submodules and restructure slightly

tfoote opened the pull request:
@sloretz This is a slightly cleaned up copy of what we looked at with the cache merged into the deploy.

tfoote committed:
Give rosdeps their own config path.

tfoote committed:
Default rosdep_path to rosdistro_path if unset in the config.

tfoote committed:
removing submodules

tfoote committed:
removing _rosfoks

tfoote committed:
Clearing out submodules and putting them into a repos file for easier management/updating.

sloretz committed:
makefile to build rosindex

Work with ros-infrastructure/container

tfoote committed:
Add deploy to resources to checkout

tfoote committed:
Move the cache inside the deploy.

sloretz commented on resources.repos:
```diff
@@ -0,0 +1,17 @@
+    version: master
+  deploy:
+    type: git
+    url: git@github.com:rosindex/rosindex.github.io.git
```
`ros-infrastructure/rosindex`?

hidmic replied:
+1 I don't think we'd want to deploy against public rosindex, though I can't seem to find said repository in `ros-infrastructure` :/

sloretz commented on _config.yml:
```diff
@@ -32,11 +32,11 @@ redcarpet:
-report_diff_filename: _caches/_deploy_index_report_diff.yaml
-debbuild_release_path: _caches/debbuild_releases
-wiki_title_index_filename: _caches/ros_org_wiki_title_index.txt
+db_cache_filename: ../deploy/cache/_deploy_cache.db
```
`checkout.sh` doesn't create the `cache` directory. Maybe touch a .gitignore file in the folder so it gets added in this PR?

sloretz replied:
I rebased #7 on top of this PR and made it make the cache directory in the `build` rule.

apojomovsky committed:
Run vcs tool during "make build"

apojomovsky committed:
Updated README.md

tfoote committed:
Move the cache out of the deploy

Inside the deploy it's getting wiped and crashing sucessive rebuilds.

tfoote committed:
Add support for site.baseurl in site links

tfoote committed:
Update site for new deployment target on github.io
set the site.baseurl and update site.url to reflect the new location
local testing can be done using the baseurl as well

tfoote committed:
Clean up a few more links for the restructure

hidmic committed:
Rebased and updated both Makefile and README.

hidmic committed:
Merge pull request #7 from ros2/deploy_to_github_pages

Deploy to GitHub pages

hidmic committed:
Fixes broken link in packages layout.

hidmic committed:
Fixes broken devel Jekyll config.

hidmic committed:
Fixes a couple broken links in layouts.

apojomovsky committed:
Fixed search

apojomovsky commented:
Just tested this along to the latest changes from [rosindex_containers#1](https://github.com/ros-infrastructure/rosindex_containers/pull/1) and is working as expected.
The rosindex submodule found in there are being updated to reflect this merge.
